.github/workflows/ci.yml

```diff
@@ -7,23 +7,51 @@ on:
 
 jobs:
   check:
-    name: check
-    runs-on: ubuntu-latest
+    name: check - ${{ matrix.platform.os_name }} with rust ${{ matrix.toolchain }}
+    runs-on: ${{ matrix.platform.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - os_name: Linux
+            os: ubuntu-latest
+          - os_name: macOS
+            os: macOS-latest
+          - os_name: Windows
+            os: windows-latest
+        toolchain:
+          - stable
+          - beta
+          - nightly
```

Comment:
Let's have `stable` and `nightly`. I think adding `beta` might be risky as it would increase chance of failures.

Reply:
Done.